src/pulserl_io.erl

```diff
@@ -110,9 +110,23 @@ decode_metadata(HeadersAndPayload) ->
             HeadersAndPayload
         end,
       <<Metadata:MetadataSize/binary, Payload/binary>> = MetadataPayload,
-      {pulsar_api:decode_msg(Metadata, 'MessageMetadata'), Payload}
+      MetaData = pulsar_api:decode_msg(Metadata, 'MessageMetadata'),
+      uncompress(MetaData, Payload)
   end.
 
+uncompress(#'MessageMetadata'{compression = 'ZLIB'} = MetaData, Payload) ->
+  Z = zlib:open(),
+  ok = zlib:inflateInit(Z),
+  [UnzippedPayload] = zlib:inflate(Z, Payload),
+  zlib:close(Z),
```

Comment:
Hey @RobKamstra!
This looks great :sparkles:.

I think de/compression is a good feature to have especially when zlib implementation is already part of the erlang OTP.
I just have one suggestion, can we move the decompression logic upward like to the connection or consumer process? That way we can keep the zlibstream open for reuse instead of opening and closing every time for every inflation.

I traced the usage of `pulserl_io:decode_metadata(HeadersAndPayload)`, where you do the inflation to https://github.com/skulup/pulserl/blob/bc6ac2fd9a52b24483d8bad75719bac4e9442d3e/src/pulserl_consumer.erl#L377

which is a handler callback to
https://github.com/skulup/pulserl/blob/bc6ac2fd9a52b24483d8bad75719bac4e9442d3e/src/pulserl_conn.erl#L349


From this standpoint it seems we've two places to do this: The connection process or the consumer process.

It seems like **_zlibstream_** per consumer process will be the better option as we may not want to block the connection process from doing IO stuffs. 

What do you think?